src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.Streaming.cs

```diff
@@ -311,6 +319,75 @@ private void WriteComponentHtml(int componentId, TextWriter output, bool allowBo
         }
     }
 
+    private void AppendWebAssemblyPreloadHeaders()
```

Comment:
Does this happen on each request ? Can we pre-compute it and reuse ?

Reply:
I agree with @pavelsavara we can do this at endpoint creation time and stash it on the endpoint metadata, similarly to how we do it for import map. See [src/Components/Endpoints/src/Builder/ResourceCollectionConvention.cs](https://github.com/dotnet/aspnetcore/pull/56045/files#diff-f603ddf3c2cac187a45e34cfc37af5d3b61ddf447509d43be9d3ec33e1f1f4b4) for details